source/detail/platform_spec_windows.cpp

```diff
@@ -114,12 +114,11 @@ namespace detail
 	platform_spec::platform_spec()
 		: impl_{ new implementation}
 	{
-		platform_abstraction::initialize();
+		platform_runtime_instance(); // force initialization
 	}
 
 	platform_spec::~platform_spec()
 	{
-		platform_abstraction::shutdown();
```

Comment:
This line is removed, because platform_runtime runtime now is a static object. The problem is that we can control the life time of the object.

The code below is the implementation of `platform_spec` in Linux
```
	platform_spec::~platform_spec()
	{
		delete msg_dispatcher_;

		//The font should be destroyed before closing display,
		//otherwise it crashs
		platform_abstraction::shutdown();

		close_display();
	}
```
We must destroy the `platform_runtime` before closing X display.

Reply:
1. I actually forgot to edit `platform_spec_posix.cpp`, right now the code will not compile.

2. In both cases (Windows, POSIX) we want `platform_spec` object to outlive `platform_runtime` object. We can easily guuarantee this by making `platform_runtime` a member of `platform_spec`.

We should not write things like `foobar::initialize()`/`foobar::shutdown()` because that's not really modern C++. It's like `new`/`delete`. What we should do instead, is to put `platform_runtime` inside `platform_spec` and the correct order of construction/destruction is guuaranteed.

Will edit appropriately.